### ISSUE TITLE
Supported runtimes not consistent

### ISSUE THREAD

**Issue by Beyley** [ManagedBass/ManagedBass#102]
It seems that what dotnet runtimes are supported is inconsistent depending on the nuget package, while the main ManagedBass package on nuget supports framework 4.5, the ManagedBass.Aac does not

**Comment by olitee** [comment]
Ah, well spotted. All the ManagedBass packages target .Net Standard 1.4, which requires .Net Framework 4.6.1 or later if you're running on .Net Framework. 

The nuget package for the base ManagedBass package does incorrectly state that it supports .Net Framework 4.5

Do you require .net 4.5 support? Or are you just letting us know about the inconsistency?

**Comment by Beyley** [comment]
I do currently require .net 4.5 support, and it does seem to actually work on .net 4.5 on my own fork that i added .net 4.5 support to (i think i had to change a few things cus of version differences tho, and its untested with .net standard 1.4, but should theoretically work with minimal changes[?])

**Comment by olitee** [comment]
I'd have to check the reasons behind targeting .NET Standard 1.4. In theory, if we lowered that target to .NET Standard 1.0, then we would officially support .NET Framework 4.5. 

@MathewSachin / @DustinBond are you guys aware of whether there is a specific reason we're targeting .NET Standard 1.4 vs 1.0?

**Comment by olitee** [comment]
@Beyley So, after some digging I've found this:

First of all - the ManagedBass nuget package does NOT compile to .NET Framework 4.5 - despite it claiming that it supports it. All the ManagedBass libraries currently compile to .NET Standard 1.4 DLLs.

ManagedBass currently depends on a number of Marshal calls to interop with the underlying Bass library. This includes calls like [Marshal.PtrToStructure](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.ptrtostructure?view=net-5.0#System_Runtime_InteropServices_Marshal_PtrToStructure_System_IntPtr_System_Type_) which was added to the CLR at .NET Framework 4.5.1, and .NET Standard 1.2.

![image](https://user-images.githubusercontent.com/1835892/131807629-d32f9e4d-ba2c-4339-993b-197ac1743889.png)

It also relies on the [File](https://docs.microsoft.com/en-us/dotnet/api/system.io.file?view=net-5.0) class, which was only added to .NET Standard at v1.3 - so .NET Standard 1.3 appears to be the lowest version we could target.

So, in short - it's **not** possible to compile ManagedBass for .NET Framework 4.5, but it **is** possible for .NET Framework 4.5.1. The risk you take running on .NET 4.5 is that any calls into the Marshall.PtrToStructure will fail with a runtime error.

How fixed to .NET 4.5 are you? Is there no possibility of bumping up to .NET 4.6, which gives you support for .NET Standard 1.3?

**Comment by Beyley** [comment]
We've tried compiling to every version of framework 4.0+ and only 4.5- work when compiled under mono then run through wine on linux (we dont have a native linux build yet but when using 4.5.0 we are able to compile under mono), if we try to compile for anything greater then 4.5.0, it crashes on startup, i can try to retarget rn as the project as last time i attempted to bump the version was months ago

**Comment by Beyley** [comment]
After reading more into dotnet standard and framework support for everything needed, it should be fine dropping down to standard 1.1, it supports the needed PtrToStructure call, and should support the oldest runtimes properly

**Comment by olitee** [comment]
@Beyley That's correct - if we switch to using some non-generic Marshall calls, then we can support .NET Standard 1.1. There are only two caveats:

1. Calls to Marshall.PtrToStructure(int,type) is marked as obsolete - but **only** for dotnetcore 1.0 and UWP. 
2. .NET Standard 1.1 doesn't support File.Exists, which is currently used for loading plugins:
https://github.com/ManagedBass/ManagedBass/blob/master/src/Bass/Shared/Bass/PInvoke/Plugin.cs#L82

So if we can figure out a workaround for that, then yes - it looks like we can scale back to .Net Standard 1.1.

**Comment by Beyley** [comment]
Very odd that File.Exists is in core 1.0 and framework 1.1 but was only added in standard 1.3, ill look into a clean replacement for it

**Comment by Beyley** [comment]
I cant find a single way of doing this, so i tried something different, which is compiling for both standard1.4 *and* framework 4.5, after changing the PtrToStructure calls, i got it compiling, ill do some testing and if it works ill make a patch file and send it here, this is probably better then whatever frankenstein method i was about to try

**Comment by Beyley** [comment]
[Add_compilation_for_dotnet_framework_4_5.patch.txt](https://github.com/ManagedBass/ManagedBass/files/7102338/Add_compilation_for_dotnet_framework_4_5.patch.txt)
this seems to work, but i havent tested it in anything other then 2 test projects, so im not sure whether it will work everywhere, but i didnt do anything breaking so i dont see why it would break

**Comment by olitee** [comment]
@Beyley thanks for raising this issue. This should now be fixed in our latest PR #130 and [4.0.0 release](https://github.com/ManagedBass/ManagedBass/releases/tag/4.0.0)
